### PR TITLE
Add Libsharp to Blue Waters, ComputeCanada, and Wheeler environments

### PR DESCRIPTION
**Pull request by nilsdeppe**
### Types of changes:

- [ ] Bugfix
- [ ] New feature

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Comment by nilsdeppe** [comment]
Marked a priority because we need to be able to compile the code.

**Comment by kidder** [comment]
@gsb76 have you verified that these work on wheeler and/or Blue Waters?

File: support/Environments/wheeler_gcc.sh [hunk 1]
```diff
@@ -38,6 +39,7 @@ spectre_load_modules() {
     module load gsl/2.1
     module load hdf5/1.8.17
     module load jemalloc/5.0.1
+    module unload libsharp/1.0.0
```
Comment (wthrowe) [inline support/Environments/wheeler_gcc.sh]: `module load`

**Review by wthrowe — APPROVED** [review]
Approving from a "code looks fine" standpoint. Someone who does work on these systems should still test it.

**Review by kidder — APPROVED** [review]

**Pull request merged**